packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -516,6 +543,13 @@ List<Error> _validateAst(Root root, String source) {
           ));
         }
       }
+      if (method.taskQueueType != TaskQueueType.serial &&
+          api.location != ApiLocation.host) {
```

Comment:
I'm confused by this check; what would it mean to explicitly set a task queue to `serial` on the other API direction? (I know it code it would be a no-op, but what would this express in the Dart API definition? Isn't the API only meaningful in one direction?)

Reply:
It's because `taskQueueType` is non-null and its default value is serial.  We don't have a way to express in the AST that it wasn't defined in the source code.